includes/data-stores/class-wc-admin-reports-customers-data-store.php

```diff
@@ -0,0 +1,399 @@
+		'date_last_active' => 'date_last_active',
+		'orders_count'     => 'orders_count',
+		'total_spend'      => 'total_spend',
+		'avg_order_value'  => 'avg_order_value',
+	);
+
+	/**
+	 * Set up all the hooks for maintaining and populating table data.
+	 */
+	public static function init() {
+		add_action( 'woocommerce_new_order', array( __CLASS__, 'sync_order' ) );
+		add_action( 'woocommerce_update_order', array( __CLASS__, 'sync_order' ) );
+		add_action( 'woocommerce_order_refunded', array( __CLASS__, 'sync_order' ) );
+	}
+
+	/**
+	 * Maps ordering specified by the user to columns in the database/fields in the data.
+	 *
+	 * @param string $order_by Sorting criterion.
+	 * @return string
+	 */
+	protected function normalize_order_by( $order_by ) {
+		if ( 'name' === $order_by ) {
+			return "CONCAT_WS( ' ', first_name, last_name )";
+		}
+
+		return $order_by;
+	}
+
+	/**
+	 * Fills ORDER BY clause of SQL request based on user supplied parameters.
+	 *
+	 * @param array $query_args Parameters supplied by the user.
+	 * @return array
+	 */
+	protected function get_order_by_sql_params( $query_args ) {
+		$sql_query['order_by_clause'] = '';
+
+		if ( isset( $query_args['orderby'] ) ) {
+			$sql_query['order_by_clause'] = $this->normalize_order_by( $query_args['orderby'] );
+		}
+
+		if ( isset( $query_args['order'] ) ) {
+			$sql_query['order_by_clause'] .= ' ' . $query_args['order'];
+		} else {
+			$sql_query['order_by_clause'] .= ' DESC';
+		}
+
+		return $sql_query;
+	}
+
+	/**
+	 * Fills WHERE clause of SQL request with date-related constraints.
+	 *
+	 * @param array  $query_args Parameters supplied by the user.
+	 * @param string $table_name Name of the db table relevant for the date constraint.
+	 * @return array
+	 */
+	protected function get_time_period_sql_params( $query_args, $table_name ) {
+		$sql_query           = array(
+			'from_clause'       => '',
+			'where_time_clause' => '',
+			'where_clause'      => '',
+		);
+		$date_param_mapping  = array(
+			'registered'  => 'date_registered',
+			'last_active' => 'date_last_active',
+			'last_order'  => 'date_last_order',
+		);
+		$match_operator      = $this->get_match_operator( $query_args );
+		$where_time_clauses  = array();
+
+		foreach ( $date_param_mapping as $query_param => $column_name ) {
+			$subclauses = array();
+			$before_arg = $query_param . '_before';
+			$after_arg  = $query_param . '_after';
+
+			if ( ! empty( $query_args[ $before_arg ] ) ) {
+				$datetime     = new DateTime( $query_args[ $before_arg ] );
+				$datetime_str = $datetime->format( WC_Admin_Reports_Interval::$sql_datetime_format );
+				$subclauses[] = "{$table_name}.{$column_name} <= '$datetime_str'";
+			}
+
+			if ( ! empty( $query_args[ $after_arg ] ) ) {
+				$datetime     = new DateTime( $query_args[ $after_arg ] );
+				$datetime_str = $datetime->format( WC_Admin_Reports_Interval::$sql_datetime_format );
+				$subclauses[] = "{$table_name}.{$column_name} <= '$datetime_str'";
```

Comment:
Shouldn't this be `"{$table_name}.{$column_name} >= '$datetime_str'";` since this is the after arg clause?

Reply:
Whoops! Copy pasta. 🍝

Reply:
Fixed in 72a7cea.